crates/core/src/constraint_system/prove.rs

```diff
@@ -654,6 +654,11 @@ where
 					<PackedType<U, FExt<Tower>>>::from_fn(|j| {
 						let index = i << log_width | j;
 
+						// If n_vars < P::LOG_WIDTH, fill the remaining scalars with zeroes.
+						if index >= 1 << n_vars {
+							return <FExt<Tower>>::ZERO;
```

Comment:
It'd be better to copy the values (repeating). That's more mathematically consistent with variable padding (repeating column).